lib/camp.js

```diff
@@ -172,6 +173,22 @@ function getCompressedStream(ask) {
   return res;
 }
 
+// Concatenate an array of streams into a single stream.
+function concatStreams(array) {
+  var concat = new stream.PassThrough();
+
+  function pipe(i) {
+    var end = i === array.length - 1;
+    array[i].pipe(concat, {end: end});
+    if (!end) {
```

Comment:
The check needs to be `if (i < array.length) { array[i].pipe(…); … }` to avoid infinite loops if array is `[]`.